src/main/java/org/agbrothers/englishbot/messagebuilder/AddWordMessageBuilder.java

```diff
@@ -0,0 +1,35 @@
+package org.agbrothers.englishbot.messagebuilder;
+
+import org.agbrothers.englishbot.entity.Word;
+import org.agbrothers.englishbot.service.DictionaryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import static org.agbrothers.englishbot.constant.ButtonLabel.*;
+
+@Component
+public class AddWordMessageBuilder extends MessageBuilder{
+    private DictionaryService dictionaryService;
+
+    @Autowired
+    public void setDictionaryService(DictionaryService dictionaryService) {
+        this.dictionaryService = dictionaryService;
+    }
+
+    @Override
+    public String getResponseMessageText(String messageText) {
+        if (messageText.equals(ADD_WORD)){
+            return "Введіть слово англійською та його переклад через пробіл";
+        }
+        String[] wordData = messageText.split(" ");
+        if(wordData.length!=2 || !wordData[0].matches("^[a-zA-Z]+$") || !wordData[1].matches("^[А-ЩЬЮЯҐЄІЇа-щьюяґєії'`’ʼ]+$")){
+            return "Помилка в слові англійською або перекладі слова. Слово англійською має бути латиницею, переклад - лише кирилицею.";
+        }
+        if(null!=dictionaryService.getWordRepository().findByEnglishValue(wordData[0])) {
+            Word wordInDictionary = dictionaryService.getWordRepository().findByEnglishValue(wordData[0]);
```

Comment:
you hit the database twice using `getWordRepository().findByEnglishValue(wordData[0])`  within this `if` statement - on lines 28 and 29
Change the code to get only one such call, do something like:
```
Word wordInDictionary = dictionaryService.getWordRepository().findByEnglishValue(wordData[0]);
if(wordInDictionary != null) 
...
```